Detectors/TRD/base/src/TrackletTransformer.cxx

```diff
@@ -0,0 +1,130 @@
+  float driftHeight = mGeo->cdrHght();
+  float x = driftHeight - 0.5;
+
+  return x;
+}
+
+float TrackletTransformer::calculateY(int hcid, int column, int position)
+{
+  double padWidth = mPadPlane->getWidthIPad();
+  int side = hcid % 2;
+
+  int nBits = constants::NBITSTRKLPOS;
+  float granularity = constants::GRANULARITYTRKLPOS;
+  int nMCMcols = constants::NCOLMCM;
+
+  // slightly modified TDP eq 16.1 (appended -1 at the end to account for pad overlap)
+  double pad = float(position - (1 << (nBits - 1))) * granularity + nMCMcols * (4 * side + column) + 10.5 - 1;
+  float y = padWidth * (pad - 72);
+
+  return y;
+}
+
+float TrackletTransformer::calculateZ(int padrow)
+{
+  double rowPos = mPadPlane->getRowPos(padrow);
+  double rowWidth = mPadPlane->getRowSize(padrow);
+  double middleRowPos = mPadPlane->getRowPos(mPadPlane->getNrows() / 2);
+
+  float z = rowPos - rowWidth / 2. - middleRowPos;
+
+  return z;
+}
+
+float TrackletTransformer::calculateDy(int slope)
+{
+  double padWidth = mPadPlane->getWidthIPad();
+
+  // temporary dummy value in cm/microsecond
+  float vDrift = 1.56;
+  float driftHeight = mGeo->cdrHght();
+
+  // dy = slope * nTimeBins * padWidth * o2::trd::constants::GRANULARITYTRKLSLOPE;
+  // nTimeBins should be number of timebins in drift region. 1 timebin is 100 nanosecond
+  double dy = slope * ((driftHeight / vDrift) / 0.1) * padWidth * constants::GRANULARITYTRKLSLOPE;
+
+  return dy;
+}
+
+float TrackletTransformer::calibrateX(double x, double t0Correction)
+{
+  return x += t0Correction;
+}
+
+float TrackletTransformer::calibrateDy(double rawDy, double oldLorentzAngle, double lorentzAngle, double driftVRatio)
+{
+  // LOG(info) << "RAW: " << rawDy;
+
+  // driftDistance = 3.35
+  float driftDistance = mGeo->cdrHght() + mGeo->camHght();
+
+  float cmSlope = rawDy / driftDistance;
+
+  double calibratedDy = rawDy - (TMath::Tan(lorentzAngle) * driftDistance);
+  calibratedDy += (TMath::Tan(oldLorentzAngle) * driftDistance * driftVRatio) + cmSlope * (driftDistance * (1 - driftVRatio));
+
+  // ALTERNATIVE METHOD
+
+  // double x_anode_hit = driftDistance*driftVRatio/cmSlope;
+  // double y_anode_hit = driftDistance*driftVRatio;
+
+  // double x_Lorentz_drift_hit = TMath::Tan(oldLorentzAngle)*driftDistance*driftVRatio - TMath::Tan(lorentzAngle)*driftDistance;
+  // double y_Lorentz_drift_hit = driftDistance*driftVRatio - driftDistance;
+
+  // double Delta_x_Lorentz_drift_hit = x_anode_hit - x_Lorentz_drift_hit;
+  // double Delta_y_Lorentz_drift_hit = y_anode_hit - y_Lorentz_drift_hit;
+  // double impact_angle_rec = TMath::ATan2(Delta_y_Lorentz_drift_hit,Delta_x_Lorentz_drift_hit);
+
+  // float calibrationShift = TMath::Tan(impact_angle_rec) * driftDistance;
+
+  // LOG(info) << "ORIGINAL: " << calibratedDy;
+  // LOG(info) << "ALTERNATIVE: " << rawDy + calibrationShift;
+
+  return calibratedDy;
+}
+
+std::vector<float> TrackletTransformer::transformL2T(int hcid, std::vector<double> point)
```

Comment:
Not sure how large the effect is, but in principle if you have a fixed size vector like in this case one would rather use a std::array<float, 3> instead of a vector.